src/cryptography/hazmat/primitives/asymmetric/rsa.py

```diff
@@ -42,13 +42,30 @@ def public_key(self):
 
 
 @six.add_metaclass(abc.ABCMeta)
-class RSAPrivateKeyWithNumbers(RSAPrivateKey):
+class RSAPrivateKeyWithSerialization(RSAPrivateKey):
```

Comment:
Fair enough; In that case this PR is once again ready for review. I added additional cross-linking and changed the name of `as_bytes` to `private_bytes` at @glyph's suggestion.